packages/bluesnap-direct-integration/src/bluesnap-direct-3ds.ts

```diff
@@ -0,0 +1,47 @@
+import {
+    guard,
+    NotInitializedError,
+    NotInitializedErrorType,
+    PaymentMethodInvalidError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import BlueSnapDirectScriptLoader from './bluesnap-direct-script-loader';
+import { BlueSnapDirectPreviouslyUsedCard, BlueSnapDirectSdk } from './types';
+
+export default class BlueSnapDirect3ds {
+    private _blueSnapSdk?: BlueSnapDirectSdk;
+
+    constructor(private _scriptLoader: BlueSnapDirectScriptLoader) {}
+
+    async initialize(testMode = false): Promise<void> {
+        this._blueSnapSdk = await this._scriptLoader.load(testMode);
+    }
+
+    async initialize3ds(
+        token: string,
+        cardData: BlueSnapDirectPreviouslyUsedCard,
+    ): Promise<string> {
+        const blueSnapSdk = this._getBlueSnapSdk();
+
+        return new Promise((resolve, reject) => {
+            blueSnapSdk.threeDsPaymentsSetup(token, (sdkResponse) => {
+                const code = sdkResponse.code;
+
+                if (code === '1') {
```

Comment:
Maybe better to create some enum or constant which will be more descriptive for this status?